scene/main/viewport.cpp

```diff
@@ -3009,6 +3009,8 @@ void Viewport::push_unhandled_input(const Ref<InputEvent> &p_event, bool p_local
 	ERR_MAIN_THREAD_GUARD;
 	ERR_FAIL_COND(p_event.is_null());
 	ERR_FAIL_COND(!is_inside_tree());
+	WARN_DEPRECATED_MSG(R"(The "push_unhandled_input" method is deprecated, use "push_input" instead.)");
```

Comment:
If it's a deprecated method, you should also wrap it in `#ifndef DISABLE_DEPRECATED`

Reply:
@YuriSizov How should I handle `#ifndef DISABLE_DEPRECATED` for the according `ClassDB::bind_method` method and the XML-doc? Wouldn't this stale XML-doc create problems, when the flag `DISABLE_DEPRECATED == true`?
I have not found another case in the code base, where an exposed function is deprecated, that I can use as a guideline for this.

Reply:
We might not have a lot of examples now as all deprecated methods should have been removed in 4.0, since we could break compatibility.

But typically you'd enclose everything related to that deprecated method in the tags, including `bind_method`, the method declaration and definition.

For the XML, you're right that it would still include info that's not relevant for a build with `deprecated=no`, but we don't consider it an issue. The editor won't use this stale info since it doesn't match anything in the ClassDB (and we don't use such builds to update the XML we commit to Git).

Reply:
Thanks for the explanation. I have updated the PR and included the `DISABLE_DEPRECATED` statements.